### PR TITLE
ask-sdk-control refactors

### PR DESCRIPTION
**Pull request by Shreyas-vgr**
## Description
- refactored default APL for ListControl
- standardized validation logic on commonControls.
- refactored all default prompts on acts to use i18n.t
- fixed error messages thrown during exceptionHandling unit test cases.

## Testing
- npm run lint success
- npm run test success

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs(Add new document content)
- [ ] Translate Docs(Translate document content)

## Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://https://github.com/alexa/ask-sdk-controls/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: src/commonControls/listControl/ListControl.ts [hunk 1]
```diff
@@ -636,9 +651,9 @@ export class ListControl extends Control implements InteractionModelContributor
             );
             const valueStr = values[0].slotValue;
             okIf(InputUtil.targetIsMatchOrUndefined(target, this.props.interactionModel.targets));
-            okIf(InputUtil.valueTypeMatch(valueType, this.props.slotType));
+            okIf(InputUtil.valueTypeMatch(valueType, this.getSlotTypes()));
             okIf(InputUtil.valueStrDefined(valueStr));
-            okIf(InputUtil.feedbackIsMatchOrUndefined(feedback, [$.Feedback.Affirm, $.Feedback.Disaffirm]));
+            okIf(InputUtil.feedbackIsMatchOrUndefined(feedback, this.getFeedBackTypes()));
```
Comment (mikelid) [inline src/commonControls/listControl/ListControl.ts:656]: nit: could be a constant I think

File: src/systemActs/ContentActs.ts [hunk 2]
```diff
@@ -69,11 +70,17 @@ export class UnusableInputValueAct<T> extends ContentAct {
...
+            controlResponseBuilder.addPromptFragment(
+                i18next.t('UNUSABLE_INPUT_VALUE_ACT_DEFAULT_PROMPT', {
+                    value: this.payload.renderedReason,
+                }),
+            );
         } else {
             throw new Error(
-                `Cannot directly render UnusableInputAct as payload.renderedReason is undefined. ${this.toString()}. ` +
-                    `Either provide a renderedReason when creating the act, or render the act in control.render() or controlManager.render()`,
+                i18next.t('CONTENT_ACT_ERROR_DEFAULT_PROMPT', {
```
Comment (mikelid) [inline src/systemActs/ContentActs.ts:80]: Errors are not usually localized. OK for these to be literals in en-US

**Review by mikelid — APPROVED** [review]
These are very helpful improvements. Thank you!

File: src/commonControls/listControl/ListControl.ts (hunk 1 above)
Comment (Shreyas-vgr) [reply src/commonControls/listControl/ListControl.ts:656]: Updated.

File: src/systemActs/ContentActs.ts (hunk 2 above)
Comment (Shreyas-vgr) [reply src/systemActs/ContentActs.ts]: Ok, reverted back 👍

**Pull request merged**